libs/wingsdk/test/target-sim/state.test.ts

```diff
@@ -5,7 +5,7 @@ import { Testing } from "../../src/simulator";
 import { State } from "../../src/target-sim";
 import { SimApp } from "../sim-app";
 
-test("state can be resolved at any time", async () => {
+test.skip("state can be resolved at any time", async () => {
```

Comment:
I basically think this capability doesn't make sense now because we are moving from a dynamic dependency resolution model to a static one and this is basically testing the dynamic model.

Are you okay if we removed this?

On a different note, I think we need a refactor around `cloud.Service`, `State`, etc. It's a bit of a mess right now.

Reply:
Yeah, I haven't pieced together yet exactly how `State` should fit into all of this either, so that's cool with me 👍

Reply:
Yes, this `State` thing is horrific. We need to get rid of it. You were right!

Reply:
Removed this test altogether for now.